### PR TITLE
feat(daemon): run init scripts

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- call ListInitScripts and execute init scripts before agent startup
- validate/order scripts by created_at/id with /bin/sh -lc execution
- add tests for parsing, ordering, and logging behavior

## Testing
- go test ./...
- go vet ./...

Refs #91

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- go test ./... (passed: 7 packages, failed: 0, skipped: 1)
- go vet ./... (no issues)

File: internal/daemon/init_scripts.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	if id == "" {
+		return initScript{}, fmt.Errorf("init script id is required")
+	}
+	createdAt := meta.GetCreatedAt()
+	if createdAt == nil {
+		return initScript{}, fmt.Errorf("init script created at is required")
+	}
+	rawScript := script.GetScript()
+	if strings.TrimSpace(rawScript) == "" {
+		return initScript{}, fmt.Errorf("init script script is required")
```
Comment (noa-lucent) [inline internal/daemon/init_scripts.go:101]: [minor] The error message `"init script script is required"` reads awkwardly due to the double "script". Consider something clearer like `"init script body is required"`.

File: internal/daemon/init_scripts.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	return initScript{
+		ID:        id,
+		CreatedAt: createdAt.AsTime(),
+		Script:    rawScript,
+	}, nil
+}
+
+func executeInitScript(ctx context.Context, script initScript, workDir string) error {
+	cmd := exec.CommandContext(ctx, "/bin/sh", "-lc", script.Script)
+	if strings.TrimSpace(workDir) != "" {
```
Comment (noa-lucent) [inline internal/daemon/init_scripts.go:112]: [minor] The `strings.TrimSpace(workDir)` check is defensive — `workDir` comes from `cfg.WorkDir` which is already resolved by the config layer. A simple `workDir != ""` would be sufficient and more consistent with how internal code should treat validated inputs. If whitespace-only paths are a concern, that validation belongs in config parsing.

File: internal/daemon/init_scripts_test.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+		CreatedAt: time.Now(),
+		Script:    "echo bad 1>&2; exit 2",
+	}
+	if err := executeInitScript(context.Background(), script, t.TempDir()); err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+	if !strings.Contains(buffer.String(), "init script script-err failed: bad") {
+		t.Fatalf("unexpected log output: %q", buffer.String())
+	}
+}
```
Comment (noa-lucent) [inline internal/daemon/init_scripts_test.go:184]: [minor] Consider adding a test that exercises `runInitScripts` (the full orchestrator) with multiple scripts where one exits non-zero. Currently `TestExecuteInitScriptNonZeroLogsAndContinues` validates the non-fatal behavior at the `executeInitScript` level, but a test through `runInitScripts` would confirm end-to-end that a failing script doesn't prevent subsequent scripts from running. Something like: two scripts where the first fails and the second writes a marker file, then assert both the log output and the marker file exist.

**Review by noa-lucent — APPROVED** [review]
Clean implementation that correctly satisfies the issue requirements:

- `ListInitScripts` with pagination ✓
- Ordering by `created_at` ASC / `id` ASC tiebreaker ✓
- Separate `/bin/sh -lc` invocation per script ✓
- Non-zero exit logs identifier + stderr, continues to next script ✓
- Runs before agent CLI spawn in all three daemon constructors ✓
- Good test coverage for parsing, ordering, and non-fatal failure ✓

Left three minor suggestions — nothing blocking. Nice work.
